### PR TITLE
refactor: Remove refresh API

### PR DESCRIPTION
**Pull request by 0sinf**
## What does this PR do?

Remove refresh API.
After check access token, if it's expired, check refresh token.
And set new token in response cookie or reject it.

## Type of change

- [ ] New feature
- [ ] Bug fix
- [x] Code improvements
- [ ] Breaking change
- [ ] Documentation
- [ ] etc.

## Changes

- Remove refresh API.
- Remove refresh token strategy
- Add check refresh token and set tokens in access token strategy
- Add response interceptor for set response cookies

### CODE REVIEW

File: src/@types/auth.d.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
 export interface JwtPayload {
   id: string;
   email: string;
-}
-
-export interface JwtPayloadWithRT extends JwtPayload {
-  refreshToken: string;
+  iat: number;
+  exp: number;
```
Comment (marshallku) [inline src/@types/auth.d.ts:5]: How about using readable names?

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -7,16 +7,22 @@ import { JwtPayload, Token } from 'src/@types/auth';
 export class AuthService {
   constructor(private jwtService: JwtService) {}
 
-  login(payload: JwtPayload): Token {
-    const accessToken = this.jwtService.sign(payload, {
-      expiresIn: '1d',
-      secret: process.env.JWT_SECRET,
-    });
+  login({ id, email }: Partial<JwtPayload>): Token {
```
Comment (marshallku) [inline src/auth/auth.service.ts]: If `id` and `email` are not optional, how about using `Pick` instead?

File: src/auth/strategies/at.strategy.ts [hunk 3]
```diff
@@ -1,23 +1,63 @@
...
-      ignoreExpiration: false,
+      ignoreExpiration: true,
+      passReqToCallback: true,
     });
   }
 
-  async validate({ id }: JwtPayload): Promise<User> {
+  async validate(req: Request, payload: JwtPayload): Promise<User> {
+    const { id } = payload;
+    const at = req.headers.authorization.replace('Bearer ', '');
```
Comment (marshallku) [inline src/auth/strategies/at.strategy.ts]: First of all, it's not time-related variable but it's named `at`.
And how about creating an utility that splits string and picks a key?

File: src/auth/strategies/at.strategy.ts [hunk 4]
```diff
@@ -1,23 +1,63 @@
...
+    const now = new Date().getTime() / 1000;
     const user = await this.userService.findById(id);
 
+    if ((decode(at) as JwtPayload).exp > now) {
+      return user;
+    }
+
+    const rt = req.cookies['refresh-token'];
```
Comment (marshallku) [inline src/auth/strategies/at.strategy.ts]: I can't get what `rt` refers to.

File: src/@types/auth.d.ts (hunk 1 above)
Comment (0sinf) [reply src/@types/auth.d.ts:5]: In type JwtPayload (jsonwebtoken), use `iat`, `exp`.

```ts
// in jsonwebtoken index.d.ts
export interface JwtPayload {
    [key: string]: any;
    iss?: string | undefined;
    sub?: string | undefined;
    aud?: string | string[] | undefined;
    exp?: number | undefined;
    nbf?: number | undefined;
    iat?: number | undefined;
    jti?: string | undefined;
}
```

Is it better that custom JwtPayload extends JwtPayload in jsonwebtoken?

Comment (marshallku) [reply src/@types/auth.d.ts:5]: Oh I think using `iat` and `exp` is better if so.

**Review by marshallku — APPROVED** [review]
LGTM!

**Pull request merged**